doc/api/async_hooks.md

```diff
@@ -0,0 +1,671 @@
+# Async Hooks
+
+
+> Stability: 2 - Stable
```

Comment:
I believe the initial release will be labeled `1 - Experimental` so that we have a major version to make any necessary tweaks.